mergeSort1.rb

```diff
@@ -8,26 +7,37 @@ def sort(arg)
 	else
 		arr_0 = arg.shift(arg.length/2)
 		arr_1 = arg
+		p arr_0
+		p arr_1
 		arr_0 = sort(arr_0)
 		arr_1 = sort(arr_1)
+		p arr_0
+		p arr_1
 	end
-	merge(arr_0, arr_1)
+	p merge(arr_0, arr_1)
 end
 
 def merge(arg, arg1)
-	puts "#{arg} #{arg1}" #Test
-	if arg[0] < arg1[0] # puts the lesser of two args
-		cont_1 = arg[0] #put the lower of the two arg into this container
-		@merged.push(cont_1) #push the containers contents to the merged array
```

Comment:
here arg and arg1 aren't being altered and then they're being passed into merge again. so merge keeps running the same code over and over without progressing so that's why it gets stuck in the infinite loop of outputing
`[1] [2]`